src/common/base/test/ArenaBenchmark.cpp

```diff
@@ -0,0 +1,90 @@
+/* Copyright (c) 2021 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include <folly/Benchmark.h>
+#include <folly/init/Init.h>
+#include <folly/memory/Arena.h>
+
+#include <string>
+#include <type_traits>
+
+#include "common/base/Arena.h"
+#include "common/expression/LabelExpression.h"
+
+namespace nebula {
+
+class TestExpr : public LabelExpression {
+ public:
+  explicit TestExpr(const std::string &name = "")
+      : LabelExpression(reinterpret_cast<ObjectPool *>(1), name) {}
+};
+
+BENCHMARK(DefaultAllocator, iters) {
+  std::size_t round = iters * 1000;
+  for (std::size_t _ = 0; _ < round; ++_) {
+    auto *expr = new TestExpr("Label");
+    delete expr;
+  }
+}
+
+BENCHMARK_RELATIVE(ArenaAllocator, iters) {
+  std::size_t round = iters * 1000;
+  Arena a;
+  for (std::size_t _ = 0; _ < round; ++_) {
+    auto *ptr = a.allocateAligned(sizeof(TestExpr));
+    auto *expr = new (ptr) TestExpr("Label");
+    expr->~TestExpr();
+  }
+}
+
+BENCHMARK_RELATIVE(FollyArenaAllocator, iters) {
+  std::size_t round = iters * 1000;
+  folly::SysArena a;
+  for (std::size_t _ = 0; _ < round; ++_) {
+    auto *ptr = a.allocate(sizeof(TestExpr));
+    auto *expr = new (ptr) TestExpr("Label");
+    expr->~TestExpr();
+  }
+}
+
+BENCHMARK_DRAW_LINE();
+
+}  // namespace nebula
+
+int main(int argc, char **argv) {
+  folly::init(&argc, &argv, true);
+
+  folly::runBenchmarks();
+  return 0;
+}
+
+// CPU info
+// Brand Raw: Intel(R) Xeon(R) CPU E5-2690 v2 @ 3.00GHz
+// Hz Advertised Friendly: 3.0000 GHz
+// Hz Actual Friendly: 3.2942 GHz
+// Hz Advertised: (3000000000, 0)
+// Hz Actual: (3294220000, 0)
+// Arch: X86_64
+// Bits: 64
+// Count: 40
+// Arch String Raw: x86_64
+// L1 Data Cache Size: 32768
+// L1 Instruction Cache Size: 32768
+// L2 Cache Size: 262144
+// L2 Cache Line Size: 256
+// L2 Cache Associativity: 6
+// L3 Cache Size: 26214400
+//
+// Build in Release mode
+//
+// ============================================================================
+// /home/shylock.huang/nebula/src/common/base/test/ArenaBenchmark.cpprelative  time/iter  iters/s
+// ============================================================================
+// DefaultAllocator                                            36.59us   27.33K
+// ArenaAllocator                                   145.89%    25.08us   39.87K
+// FollyArenaAllocator                              138.96%    26.33us   37.98K
```

Comment:
Goog job! We should also update the parser benchmarks.

Reply:
Benchmark of origin:
```
============================================================================
/home/shylock.huang/nebula-tmp2/src/parser/test/ParserBenchmark.cpprelative  time/iter  iters/s
============================================================================
SimpleQuery(1_thread)                                        1.20us  836.34K
SimpleQuery(2_thread)                             96.66%     1.24us  808.44K
SimpleQuery(4_thread)                             94.08%     1.27us  786.79K
SimpleQuery(8_thread)                             94.90%     1.26us  793.64K
SimpleQuery(16_thread)                            95.07%     1.26us  795.14K
SimpleQuery(32_thread)                            66.19%     1.81us  553.59K
SimpleQuery(48_thread)                            47.52%     2.52us  397.44K
----------------------------------------------------------------------------
ComplexQuery(1_thread)                                       8.61us  116.12K
ComplexQuery(2_thread)                            97.47%     8.84us  113.18K
ComplexQuery(4_thread)                            95.44%     9.02us  110.83K
ComplexQuery(8_thread)                            97.86%     8.80us  113.64K
ComplexQuery(16_thread)                           98.14%     8.78us  113.96K
ComplexQuery(32_thread)                           95.71%     9.00us  111.14K
ComplexQuery(48_thread)                           60.94%    14.13us   70.76K
----------------------------------------------------------------------------
MatchConflictQuery(1_thread)                                 3.65us  274.17K
MatchConflictQuery(2_thread)                      96.74%     3.77us  265.23K
MatchConflictQuery(4_thread)                      94.19%     3.87us  258.25K
MatchConflictQuery(8_thread)                      93.17%     3.91us  255.45K
MatchConflictQuery(16_thread)                     94.70%     3.85us  259.64K
MatchConflictQuery(32_thread)                     91.22%     4.00us  250.10K
MatchConflictQuery(48_thread)                     57.08%     6.39us  156.50K
============================================================================
```

Benchmark of current:
```
============================================================================
/home/shylock.huang/nebula-tmp/src/parser/test/ParserBenchmark.cpprelative  time/iter  iters/s
============================================================================
SimpleQuery(1_thread)                                        1.07us  931.55K
SimpleQuery(2_thread)                             98.63%     1.09us  918.82K
SimpleQuery(4_thread)                             97.50%     1.10us  908.26K
SimpleQuery(8_thread)                             96.84%     1.11us  902.09K
SimpleQuery(16_thread)                            93.79%     1.14us  873.73K
SimpleQuery(32_thread)                            65.51%     1.64us  610.22K
SimpleQuery(48_thread)                            41.94%     2.56us  390.73K
----------------------------------------------------------------------------
ComplexQuery(1_thread)                                       8.34us  119.88K
ComplexQuery(2_thread)                            98.58%     8.46us  118.18K
ComplexQuery(4_thread)                            96.30%     8.66us  115.45K
ComplexQuery(8_thread)                            92.82%     8.99us  111.27K
ComplexQuery(16_thread)                           93.72%     8.90us  112.36K
ComplexQuery(32_thread)                           94.08%     8.87us  112.78K
ComplexQuery(48_thread)                           59.36%    14.05us   71.16K
----------------------------------------------------------------------------
MatchConflictQuery(1_thread)                                 3.83us  261.02K
MatchConflictQuery(2_thread)                      97.54%     3.93us  254.61K
MatchConflictQuery(4_thread)                      99.23%     3.86us  259.00K
MatchConflictQuery(8_thread)                      96.97%     3.95us  253.10K
MatchConflictQuery(16_thread)                     96.94%     3.95us  253.03K
MatchConflictQuery(32_thread)                     91.72%     4.18us  239.40K
MatchConflictQuery(48_thread)                     57.47%     6.67us  150.01K
============================================================================
```